### ISSUE TITLE
Rule Request: AvoidSimpleKeyCollisions

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#152]
for using a SimpleKey in a keyGenerator, violate when 
[classname and/or method name -> method]
is missing, a rule to check this.

Results of two methods with both only relationId as argument will get mixed up:
```java
    @Cacheable(value="DATA", sync = true, keyGenerator = "cacheKeyGenerator") 
    public Object getLimitedData(String id) {
        return fetchLimitedFromBackend(id);
    }
   @Cacheable(value="DATA", sync = true, keyGenerator = "cacheKeyGenerator") 
   public Object getExtendedData(String id) {
        return fetchExtendedFromBackend(id);
    }
```

**Comment by jborgers** [comment]
```
import org.springframework.cache.interceptor.KeyGenerator;
import org.springframework.cache.interceptor.SimpleKey;
import java.lang.reflect.Method;

class GoodCacheKeyGenerator implements KeyGenerator {
    @Override
    public Object generate(Object target, Method method, Object... params) {
        return new SimpleKey(target.getClass().getName(), method.getName(), params);
    }
}
class BadCacheKeyGenerator implements KeyGenerator {
    @Override
    public Object generate(Object target, Method method, Object... params) {
        return new SimpleKey(params);
    }
}

```
